### DISCUSSION TITLE
Forces on Moving Boundaries

### DISCUSSION THREAD

**Discussion by ncalbrecht** [ProjectPhysX/FluidX3D#139] [Q&A]
Hi,
first up thanks for the impressive software! I want to figure out, wether I can use FluidX3D to estimate the thrust of a propeller. I know, the current LBM implementation has issues with highly turbulent flow and needs very fine discretisation to avoid stairstepping, but I wanted to try what happens if I just crank up the mesh to sub mm size in water. However, as soon as I use moving boundaries for my propeller objects, the forces are always reported to 0. Force calculation for static objects works reasonable, but as soon as I add velocity during voxelisation it just reports 0.  Is this intended behaviour or am I doing something wrong?

**Comment by ProjectPhysX** [comment]
I don't think this is possible with boundary forces directly. Better infer thrust force from generated flux (from velocity field) through an area behind the propeller.
Long answer here: https://github.com/ProjectPhysX/FluidX3D/issues/141#issuecomment-2045970307
